app/controllers/order/international_order.js

```diff
@@ -0,0 +1,141 @@
+import Ember from "ember";
+import SearchOptionMixin from "stock/mixins/search_option";
+import AsyncMixin, { ERROR_STRATEGIES } from "stock/mixins/async";
+import { regex } from "stock/constants/regex";
+
+export default Ember.Controller.extend(SearchOptionMixin, AsyncMixin, {
+  i18n: Ember.inject.service(),
+  showError: false,
+  orderService: Ember.inject.service(),
+
+  shipmentorCarryoutCode: Ember.computed({
+    get() {
+      return this.get("shipmentorCarryoutCode");
+    },
+    set(_, value) {
+      return value;
+    }
+  }),
+
+  isInvalidDescription: Ember.computed("orderDescription", function() {
+    return !this.get("orderDescription");
+  }),
+
+  isInvalidCode: Ember.computed("shipmentorCarryoutCode", function() {
+    const codeRegEx = new RegExp(regex.SHIPMENT_ORDER_REGEX);
+    return (
+      !this.get("shipmentorCarryoutCode") ||
+      !codeRegEx.test(this.get("shipmentorCarryoutCode"))
+    );
+  }),
+
+  isInValidCountry: Ember.computed("country", function() {
+    return !this.get("country");
+  }),
+
+  isInvalidShipmentDate: Ember.computed("shipmentDate", function() {
+    return !this.get("shipmentDate");
+  }),
+
+  getCurrentUser: Ember.computed(function() {
```

Comment:
I think there already exist `getCurrentUser` in service, if not create one and you directly access service method in `.hbs`.

Reply:
yup.
can make use of app/service/session.js